scrapinghub/client/exceptions.py

```diff
@@ -69,6 +73,8 @@ def wrapped(*args, **kwargs):
                 raise ValueTooLarge(http_error=exc)
             elif 400 <= status_code < 500:
                 raise ScrapinghubAPIError(http_error=exc)
+            elif 500 <= status_code < 600:
+                raise ServerError(http_error=exc)
```

Comment:
That's a fair point, thanks. Return code is 403 for the authentication errors, so just calling `raise_for_status` doesn't fit. There's a few options how to handle it more gracefully in my mind:

- always raise HTTPError for all 500s errors(500<=status<600), w/o trying to parse body
- wrap `json.loads` with try-except and raise HTTPError from except clause (assuming that successful requests' responses always have proper body type, i.e. json)

My guess that 2nd approach is a way to go, and we also should do the same for jsonlines case. And as `ValueError` can happen in `_decode_response` on `json.loads` only, I'd wrap the whole method by catching `ValueError`s and raising `HTTPError`s based on response. Do you agree or I'm missing something?